src/python_testing/TC_EWATERHTR_2_3.py

```diff
@@ -0,0 +1,285 @@
+                     "Verify Command response is Success"),
+            TestStep("10a", "TH reads HeatDemand attribute.",
+                     "Verify value is 0x00 (no demand on any source)"),
+            TestStep("10b", "TH reads BoostState attribute.",
+                     "Verify value is 1 (Active)"),
+            TestStep("10c", "TH reads TankPercentage attribute.",
+                     "Verify value is 75%"),
+            TestStep("11", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER for Draw off hot water Test Event.",
+                     "Verify Command response is Success"),
+            TestStep("11a", "TH reads HeatDemand attribute.",
+                     "Verify value is greater than 0x00 (demand on at least one source) and (HeaterDemand & (!HeaterTypes)) is zero (demand is only from declared supported types)"),
+            TestStep("11b", "TH reads BoostState attribute.",
+                     "Verify value is 1 (Active)"),
+            TestStep("11c", "TH reads TankPercentage attribute.",
+                     "Verify value is 50%"),
+            TestStep("12", "TH sends CancelBoost.",
+                     "Verify Command response is Success"),
+            TestStep("12a", "TH reads HeatDemand attribute.",
+                     "Verify value is 0x00 (no demand on any source)"),
+            TestStep("12b", "TH reads BoostState attribute.",
+                     "Verify value is 0 (Inactive)"),
+            TestStep("12c", "TH reads TankPercentage attribute.",
+                     "Verify value is 50%"),
+            TestStep("13", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER for Basic installation Test Event Clear.",
+                     "Verify Command response is Success"),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_3(self):
+
+        self.step("1")
+        # Commission DUT - already done
+
+        self.step("2")
+        await self.check_test_event_triggers_enabled()
+
+        self.step("3")
+        await self.send_test_event_trigger_basic_installation_test_event()
+
+        self.step("3a")
+        await self.check_whm_attribute("HeatDemand", 0)
+
+        self.step("3b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
+
+        self.step("3c")
+        await self.check_whm_attribute("TankPercentage", 0)
```

Comment:
Shouldn't there be a PICS guard here since TankPercentage is not a mandatory attribute?